resources/projectsyncup-design-document.md

```diff
@@ -0,0 +1,199 @@
+
+### 6.4. _Create Task Endpoint_
+* Accepts `POST` requests to `/tasks`
+* Accepts data to create a new Task with a given task name, description, due date, task status, and list of assigned team members. Returns the new Task, including a unique task ID assigned by the Project Management Service.
+* Request Body: <br>
+    ```{"taskName": "Task Title",``` <br>
+    ```"taskDescription": "Task Description", ```<br>
+    ```"taskStatus": "Not Started", ```<br>
+    ```"taskAssignedUsers": ["userId_1", "userId_2"], ```<br>
+    ```"taskStatus": "open", ```<br>
+    ```"taskDueDate": "yyyy-mm-dd", ```<br>
+    ```"taskTimer": 60```<br>
+
+![](images/create-task-activity-SD.png)
+
+### 6.5. _Add Task To Project Endpoint_
+* Accepts `PUT` requests to `/projects/:id/task`
+* Accepts a project ID and task ID and adds the task to the specified project. 
+  * If the project is not found, will throw a ProjectNotFoundException 
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+### 6.6. _Update Task Information Endpoint_
+* Accepts `PUT` requests to `/tasks/:id`
+* Accepts a task ID and updated task information, Returns the updated the task with the new information.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+* Request Body: <br>
+  ```{"taskName": "Task Title: New Title",``` <br>
+  ```"taskDescription": " Updated Task Description", ```<br>
+  ```"taskAssignedUsers": ["userId_1", "userId_2"], ```<br>
+  ```"taskStatus": "Not Started", ```<br>
+  ```"taskDueDate": "yyyy-mm-dd", ```<br>
+  ```"taskTimer": 60```<br>
+
+### 6.7. _Update Task Progress Endpoint_
+* Accepts `PUT` requests to `/tasks/:id/progress`
+* Accepts a task ID and a new progress status (Not Started, In Progress, Complete), updates the ``taskStatus`` of the task with that ID.
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+* Request Body: <br>
+    `{ "status": "Complete" }`
+
+### 6.8. _Track Time Spent on Task Endpoint_
+* Accepts `PUT` requests to `/tasks/:id/time`
+* Accepts a task ID and a new time spent on task, updates the ``taskTimer`` of the task with that ID.
+  * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+* Request Body: <br>
+    `{ "taskTimer": 61 }` 
+
+### 6.9. _Delete Task From Project Endpoint_
+* Accepts `DELETE` requests to `/tasks/:id`
+* Accepts a task ID, deletes the task with that ID from project.
+    * If the given task ID doesn't exist, will throw an ``TaskNotFoundException``
+
+### 6.10. _View All Projects Endpoint_
+* Accepts `GET` requests to `/projects/:id`
+* Accepts a project ID, returns a list of projects created by that Manager.
+    * If the given project ID doesn't exist, will throw an ``ProjectNotFoundException``
+    * If the given Manager has not created any projects, an empty list will be returned
+
+### 6.11. _View All Assigned Tasks Endpoint_
+* Accepts `GET` requests to ` /tasks/assigned/:id`
```

Comment:
You'll be able to get the users ID from Cognito, so don't need to specify it as a URL parameter here.